lib/validate_style.js

```diff
@@ -0,0 +1,47 @@
+'use strict';
+
+var validateConstants = require('./validate/validate_constants');
+var validate = require('./validate/validate');
+var latestStyleSpec = require('../reference/latest.min');
+var parseStyle = require('./parse_style');
```

Comment:
It sucks that our file structure is a "public API" :disappointed: 

Is this an acceptable compromise, @jfirebaugh?

 - add a shim for `mapbox-gl-style-spec/lib/validate/latest` https://github.com/mapbox/mapbox-gl-style-spec/commit/405a3389d80464c0b5f4b6e009841d5c2217b51c
 - add an explicitly minified style validation entrypoint `mapbox-gl-style-spec/lib/validate_style.min` https://github.com/mapbox/mapbox-gl-style-spec/commit/d059087beb1f3cfcc1c38841d1e241457473446b

If so, I will

1. documentation.js the _heck_ out of this module
2. build a solution for https://github.com/mapbox/mapbox-gl-js/issues/1632
3. merge this into master